fuel-merkle/test-helpers/src/binary/merkle_tree.rs

```diff
@@ -88,7 +89,14 @@ impl MerkleTree {
             current = current.take_next().unwrap();
         }
 
-        (self.root(), self.proof_set)
+        let root = self.root();
+
+        // The proof set starts with the hashed leaf data; remove the leaf data
+        // in order to conform with the proof set specification.
+        let mut proof_set = VecDeque::from(self.proof_set);
+        proof_set.pop_front();
```

Comment:
This removes the hashed leaf data from the `test-helpers` reference version of the BMT proof generation. The reference Merkle tree algorithm used here is different than the one used in production. Because of this, we can only remove the leaf data from the proof after the proof is generated.